files/aws_lifecycle_hooks/tools.py

```diff
@@ -74,19 +116,128 @@ def get_asg_name(
         instance_id: str,
         asg_client,
 ) -> str:
-    asg_info = asg_client.describe_auto_scaling_instances(InstanceIds=[instance_id])
-    asg_name = asg_info[u'AutoScalingInstances'][0][u'AutoScalingGroupName']
-    print("I am part of Auto Scaling Group {asg_name}".format(asg_name=asg_name))
+    """Use describe asg to get asg group name."""
+    import botocore.exceptions
+    asg_name = None
+    try:
+        asg_info = asg_client.describe_auto_scaling_instances(InstanceIds=[instance_id])
+        asg_name = asg_info[u'AutoScalingInstances'][0][u'AutoScalingGroupName']
+        print("I am part of Auto Scaling Group {asg_name}".format(asg_name=asg_name))
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == "AccessDenied":
+            print('This instance/user has no permission to query autoscaling.')
```

Comment:
This branch returns None, if this is the intended behaviour, you should update the type hint to return `typing.Optional[str]`